experimental/packages/opentelemetry-api-metrics/src/types/Metric.ts

```diff
@@ -88,21 +89,12 @@ export interface Histogram {
   record(value: number, attributes?: Attributes, context?: Context): void;
 }
 
-// ObservableBase has to be an Object but for now there is no field or method
-// declared.
-/** Base interface for the Observable metrics. */
-export type ObservableBase = Record<never, never>;
-
-/** Base interface for the ObservableGauge metrics. */
-export type ObservableGauge = ObservableBase;
-
-/** Base interface for the ObservableUpDownCounter metrics. */
-export type ObservableUpDownCounter = ObservableBase;
-
-/** Base interface for the ObservableCounter metrics. */
-export type ObservableCounter = ObservableBase;
-
 /**
  * key-value pairs passed by the user.
  */
 export type Attributes = { [key: string]: string };
+
+/**
+ * The observable callback for Observable instruments.
+ */
+export type ObservableCallback = (observableResult: ObservableResult) => void | Promise<void>;
```

Comment:
Is `ObservableResult` the name used in spec? I know it was the name used previously in JS but to me seems like a confusing name. Might be a good time now to come up with a better one? Maybe `Observer` would be more appropriate?

Reply:
Yes, the name "ObservableResult" is used in the spec:

> OpenTelemetry API authors MAY decide what is the idiomatic approach. Here are some examples:
> 1. Return a list (or tuple, generator, enumerator, etc.) of Measurements.
> 2. Use an observer result argument to allow individual Measurements to be reported.
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-counter-creation

I checked out other implementations and found that:
1. Java: with the parameter pattern and the parameter is named as "Observable*Measurement": https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/metrics/ObservableDoubleMeasurement.java#L11;
2. Go: seems outdated: https://github.com/open-telemetry/opentelemetry-go/blob/main/metric/metric_instrument.go#L31
3. DotNet: seems like named as MetricListener but I'm not sure I understand the implementation correctly 😅  https://github.com/open-telemetry/opentelemetry-dotnet/blob/metrics/src/OpenTelemetry/Metrics/MeterProviderSdk.cs#L74;
4. Python: seems not implemented yet, in the prototype PR it seems like it is implemented in the return value pattern (with iterator/generator): https://github.com/open-telemetry/opentelemetry-python/pull/2280/files#diff-e9e09d3bf2bfab2ac424002d5c3519755e4a5fbca6927306815c45cac07b5647R11

So with the spec compliance, we have the following options:

1. Name the parameter as "ObservableResult", not changing,
3. Return a measurement list from the ObservableCallback,
4. Return an async generator from the ObservableCallback.

Reply:
FWIW, I submitted a PR to fix the term "observer result" with "observable result": https://github.com/open-telemetry/opentelemetry-specification/pull/2238